addon/-private/service-injections.js

```diff
@@ -1,15 +1,25 @@
-const SERVICE_INjECTIONS = new WeakMap();
+const SERVICE_INJECTIONS = new WeakMap();
+const RESOLVED_SERVICE_INJECTIONS = new WeakMap();
 
 export function setServiceInjections(fn, services = []) {
-  SERVICE_INjECTIONS.set(fn, services);
+  SERVICE_INJECTIONS.set(fn, services);
 }
 
 export function getServiceInjections(fn, owner) {
-  if (!SERVICE_INjECTIONS.has(fn)) {
+  if (!SERVICE_INJECTIONS.has(fn)) {
     return [];
   }
 
-  const services = SERVICE_INjECTIONS.get(fn);
+  if (RESOLVED_SERVICE_INJECTIONS.has(fn)) {
```

Comment:
This is subtly wrong, and would cause issues with using a prior tests service. 

In my example snippet from https://github.com/spencer516/ember-functional-modifiers/pull/3/commits/0707c54a61a69ea21cff7106de70a2a4cbd00397#r261842187, I added a single `this.servicesCache` to the instance of `FunctionModifierManager`, however _because_ that instance itself is inherently tied to the owner already its "fine". In the implementation here at the moment, the caching is **global** not tied to the owner. Since each test gets a new owner (as well as each fastboot request), the single layer caching done in this util method would mean getting an old owner's services (which would be bad....).

Reply:
Oops — I hadn't seen that comment from above! Awesome. :)